web/src/main/java/com/synopsys/integration/alert/web/api/provider/project/ProviderProjectCustomFunctionAction.java

```diff
@@ -60,23 +67,48 @@ public ProviderProjectCustomFunctionAction(AuthorizationManager authorizationMan
 
     @Override
     public ActionResponse<ProviderProjectOptions> createActionResponse(FieldModel fieldModel, HttpServletContentWrapper servletContentWrapper) {
+        HttpServletRequest httpRequest = servletContentWrapper.getHttpRequest();
+        Map<String, String[]> parameterMap = httpRequest.getParameterMap();
+
+        int pageNumber = extractPagingParam(parameterMap, "pageNumber", 0);
+        int pageSize = extractPagingParam(parameterMap, "pageSize", 10);
```

Comment:
We don't have to change this in this version of Alert but I think this is a good argument for getting rid of the dynamic endpoints and just create controllers/endpoints as we need. Having to extract the parameters this way is not appealing.

Reply:
Agreed. I think it will be part of a larger effort to rewrite the Jobs UI to have tightly coupled channel endpoints.